### DISCUSSION TITLE
Add Oracle Connector

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [evidence-dev/evidence#2340] [Data Source Request]
As in #5 it would be nice to have connectors for other dbs. Or maybe just a generic `CustomConnector` that can be any of available db connectors on npm.

**Comment by archiewood** [comment]
Hello -looping back on connector requests

We have recently re-architected Evidence (since this issue was opened), and Evidence is now "pluggable", meaning you can write your own connectors, publish them, and use them in Evidence.

If you are interesting in writing a connector, here are some resources:

1. Docs on creating a plugin: https://docs.evidence.dev/plugins/creating-a-plugin/datasources
2. Template for a connector plugin: https://github.com/evidence-dev/datasource-template
3. An example that I spun up for pulling stats from the npm API: https://github.com/archiewood/npm-stats

If you do write a plugin:

- the suggested package name is `evidence-connector-[plugin]`
- the docs are new, so let us know if there is extra information you need
- share it on the Evidence [community](https://github.com/evidence-dev/evidence/issues/slack.evidence.dev), and we'll feature it!

**Comment by rinie** [comment]
Saw the naming suggestions just now...
I implemented a https://www.npmjs.com/package/evidence-connector-oracledb
Or https://github.com/rinie/evidence-connector-oracledb
Mostly based on the Postgres connector

It is not clear to me how I could initialize the driver once for thick mode...
But I get some results from an Oracle database...

sql in the frontend is nice.

**Comment by archiewood** [comment]
Very cool! 

PR a link into awesome-evidence to make it community discoverable?

https://github.com/evidence-dev/awesome-evidence

**Comment by rinie** [comment]
Done

**Comment by rinie** [comment]
Trying to use evidence v39 and Node v22 on my previous setup.
Now documentation suggest it should be databases instead of datasources?
And I get 'Datasource package "evidence-connector-oracledb" is an invalid Evidence plugin' of invalid in de settings view
